sdcm/nemesis.py

```diff
@@ -61,6 +67,99 @@ class UnsupportedNemesis(Exception):
     """ raised from within a nemesis execution to skip this nemesis"""
 
 
+class FieldTypeDefinition():
+    ALL_TYPES = ('ascii', 'blob', 'counter', 'decimal', 'duration', 'int', 'map', 'smallint', 'time',
+                 'timeuuid', 'uuid', 'varint', 'bigint', 'boolean', 'date', 'double', 'float', 'inet', 'list', 'set',
+                 'text', 'timestamp', 'tinyint', 'varchar')
+    NOT_EMBEDDABLE_TYPES = ['duration', 'counter']
+    FROZEN_EMBEDDABLE_TYPES = ['set', 'list', 'map']
+    maintype = None
+    subtype1 = None
+    subtype2 = None
+    parent = None
+
+    def __init__(self, maintype=None, subtype1=None, subtype2=None):
+        if maintype:
+            self.maintype = maintype
+        if subtype1:
+            self.subtype1 = subtype1
+        if subtype2:
+            self.subtype2 = subtype2
+
+    @classmethod
+    def get_random(cls, already_created_info, avoid_types=None, max_level=1, allowed_types=None):
+        if avoid_types is None:
+            avoid_types = []
+        if allowed_types is None:
+            allowed_types = cls.ALL_TYPES
+        excluded_types = set(avoid_types + list(already_created_info.keys()) + ['set', 'list', 'map'])
+        real_allowed_types = [e for e in allowed_types if e not in excluded_types]
+        if real_allowed_types:
+            while True:
+                maintype = random.choice(real_allowed_types)
+                if maintype not in avoid_types:
+                    break
```

Comment:
choosen maintype will not be in avoid types always. May be it will be enough only  
`maintype = random.choice(real_allowed_types) `

Reply:
Very true, fixed.